src/queue.rs

```diff
@@ -248,73 +222,187 @@ pub fn pop<'d, S: MultiwriteNorFlash>(
     flash_range: Range<u32>,
     data_buffer: &'d mut [u8],
 ) -> Result<Option<&'d mut [u8]>, Error<S::Error>> {
-    assert_eq!(flash_range.start % S::ERASE_SIZE as u32, 0);
-    assert_eq!(flash_range.end % S::ERASE_SIZE as u32, 0);
+    pop_many(flash, flash_range).next(data_buffer)
+}
 
-    assert!(S::ERASE_SIZE >= S::WORD_SIZE * 4);
-    assert!(S::WORD_SIZE <= MAX_WORD_SIZE);
+/// Iterator for pop'ing elements in the queue.
+pub struct PopIterator<'d, S: MultiwriteNorFlash> {
+    iter: QueueIterator<'d, S>,
+}
+
+impl<'d, S: MultiwriteNorFlash> PopIterator<'d, S> {
+    /// Pop the next event.
+    ///
+    /// The data is written to the given `data_buffer`` and the part that was written is returned.
```

Comment:
```suggestion
    /// The data is written to the given `data_buffer` and the part that was written is returned.
```

(Apparently this is my fault and you just copied it. Could you fix this in the original `pop` and `peek` too? I can't place a suggestion on code you didn't change... :/)